src/lib/dnssd/Advertiser_ImplMinimalMdns.cpp

```diff
@@ -670,10 +671,10 @@ CHIP_ERROR AdvertiserMinMdns::Advertise(const CommissionAdvertisingParameters &
         }
     }
 
-    if (params.GetVendorId().HasValue())
+    const auto vendorId = params.GetVendorId();
```

Comment:
Doesn't this make a copy?  Do we want that?  Or should this be `const auto &`?

Reply:
The return value was `chip::optional<...>` and now it is std::optional ... both without reference. I think we can treat small optionals the same as just simple values and copy by value. Only structures may be worth it to reference. Looking for tnose now, assuming MRPConfig falls under that only.